distributed/worker.py

```diff
@@ -1519,11 +1515,15 @@ def add_task(
 
                     # transition from new -> fetch handles adding dependency
                     # to waiting_for_data
-                    self.transition(dep_ts, state)
+                    # TODO: `state` should be "fetch" and let recommender
+                    # make it "memory" if needed
+                    recommendations.update({dep_ts: state})
```

Comment:
The code that defines `state` isn't visible in my view here, but this current check is for handling tasks that get re-added, e.g. we've nuked the corresponding `TaskState` object but the _result_ still lives in `self.data`

This would go away (one hopes) if we stop removing `TaskState` instances.